packages/wrangler/src/api/startDevWorker/BaseController.ts

```diff
@@ -1,95 +1,53 @@
-import { EventEmitter } from "node:events";
 import { logger } from "../../logger";
 import type {
 	BundleCompleteEvent,
 	BundleStartEvent,
+	ConfigUpdateEvent,
 	DevRegistryUpdateEvent,
 	ErrorEvent,
 	PreviewTokenExpiredEvent,
 	ReloadCompleteEvent,
 	ReloadStartEvent,
 } from "./events";
 
-interface TypedEventEmitter<EventMap extends Record<string | symbol, unknown[]>>
-	extends EventEmitter {
-	addListener<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	on<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	once<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	removeListener<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	off<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	removeAllListeners(event?: keyof EventMap): this;
-	listeners<Name extends keyof EventMap>(
-		eventName: Name
-	): ((...args: EventMap[Name]) => void)[];
-	rawListeners<Name extends keyof EventMap>(
-		eventName: Name
-	): ((...args: EventMap[Name]) => void)[];
-	emit<Name extends keyof EventMap>(
-		eventName: Name,
-		...args: EventMap[Name]
-	): boolean;
-	listenerCount<Name extends keyof EventMap>(
-		eventName: Name,
-		listener?: (...args: EventMap[Name]) => void
-	): number;
-	prependListener<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-	prependOnceListener<Name extends keyof EventMap>(
-		eventName: Name,
-		listener: (...args: EventMap[Name]) => void
-	): this;
-}
+export type ControllerEvent =
+	| ErrorEvent
+	| ConfigUpdateEvent
+	| BundleStartEvent
+	| BundleCompleteEvent
+	| ReloadStartEvent
+	| ReloadCompleteEvent
+	| DevRegistryUpdateEvent
+	| PreviewTokenExpiredEvent;
 
-const TypedEventEmitterImpl = EventEmitter as unknown as {
-	new <
-		EventMap extends Record<string | symbol, unknown[]>,
-	>(): TypedEventEmitter<EventMap>;
-};
+export interface ControllerBus {
+	dispatch(event: ControllerEvent): void;
+}
 
-export type ControllerEventMap = {
-	error: [ErrorEvent];
-};
-export abstract class Controller<
-	EventMap extends ControllerEventMap = ControllerEventMap,
-> extends TypedEventEmitterImpl<EventMap> {
+export abstract class Controller {
+	protected bus: ControllerBus;
 	#tearingDown = false;
+
+	constructor(bus: ControllerBus) {
+		this.bus = bus;
+	}
```

Comment:
Would this work? Plus removing the definition on line 28

```suggestion
	constructor(protected bus: ControllerBus) {
	}
```